st3/lsp_utils/_client_handler/language_handler.py

```diff
@@ -144,3 +148,24 @@ def __init__(self):
         # Will be a no-op if already ran.
         # See https://github.com/sublimelsp/LSP/issues/899
         self.setup()
+
+    def _register_custom_server_event_handlers(self, api: ApiWrapperInterface) -> None:
```

Comment:
Does it need to be done separately for each handler implementation. Did the previous approach of changing generic client handler didn't work?

Reply:
It works but potentially error-prone if people forget to do `super().on_ready()` in their custom `on_ready()` then `_register_custom_server_event_handlers()` wont be executed.

Reply:
But on the other-hand such duplication is worse. The big goal of the refactor was to get rid of as much duplication as possible.

You can add some internal functions to the client interface that will be called by handler implementations before `on_ready` and which generic handler can decorate from.

Reply:
I guess https://github.com/sublimelsp/lsp_utils/pull/38/commits/b9f81229128f1780ba2c29de4ff56175abfd9f6b resolves this at a certain degree.